pull_request_template.md

```diff
@@ -0,0 +1,22 @@
+<!--
+Resources:
+* contributing guidelines: https://catalystcoop-pudl.readthedocs.io/en/latest/CONTRIBUTING.html
+* code of conduct: https://catalystcoop-pudl.readthedocs.io/en/latest/code_of_conduct.html
+-->
+# Overview
+
+Closes #XXXX.
+
+What problem does this address?
+
+What did you change?
+
+# Testing
+
+How did you make sure this worked? How can a reviewer verify this?
+
+```[tasklist]
+# To-do list
```

Comment:
Right now the task list looks like it's nested under "testing" Maybe could have a #Todo section?